src/net/irlap.c

```diff
@@ -147,7 +154,13 @@ struct irlap_cb *irlap_open(struct net_device *dev, struct qos_info *qos,
 	} while ((self->saddr == 0x0) || (self->saddr == BROADCAST) ||
 		 (hashbin_lock_find(irlap, self->saddr, NULL)) );
 	/* Copy to the driver */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
 	memcpy((void *)dev->dev_addr, &self->saddr, 4);
+#else
+	memset(eth_addr, 0, sizeof(eth_addr));
+	memcpy(eth_addr, &self->saddr, MIN(sizeof(eth_addr), sizeof(self->saddr)));
+	eth_hw_addr_set(dev, eth_addr);
```

Comment:
`eth_hw_addr_set` is the wrong abstraction here.

I think this should rather be just `__dev_addr_set(dev, &self->saddr, LAP_ALEN);`

Reply:
agreed. the latest change uses dev_addr_set(), which uses dev->addr_len. that should have been set to LAP_ALEN, so just BUG() if that's not the case.